Sources/sourcekit-lsp/SourceKitLSP.swift

```diff
@@ -193,6 +193,11 @@ struct SourceKitLSP: AsyncParsableCommand {
   )
   var generatedInterfacesPath = defaultDirectoryForGeneratedInterfaces
 
+  @Option(
+    help: "Specify the directory where generated macro expansions will be stored"
+  )
+  var generatedMacroExpansionsPath = defaultDirectoryForGeneratedMacroExpansions
+
```

Comment:
Instead of adding options for every new kind of generated file directories, what do you think about generalizing `generatedInterfacesPath` to `generatedFilesPath` and then placing the generated interfaces and the macro expansions in those subdirectories?